vis.py: remove filter rows parameter from vis functions

kaushik-rohit opened the pull request:
Currently many functions in vis have intial_row_filter
which is being used to determine the data to be used
for plotting. This leaves vis doing more work than just
plotting. The data should be preprocessed outside the vis
and not inside.

Functions with filter ranges:
1) generate_probability_distribution
2) exponential_curve_fit_and_plot
3) exponential_curve_fit_and_plot_x_shifted
4) generate_log_plots
This commit modifies all the above functions and
remove filter parameter.
Rather new functions will be added to determine the correct
percentile cutoff for data selection.

Issue Link: #136
Issue Link: #141

## What? Why?
Fix #136 #141 

Changes proposed in this pull request:
Remove upper limit parameters from function in vis.py

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Any images?
![cc_degree_curve_fit](https://user-images.githubusercontent.com/22174615/30583583-f959f22c-9d16-11e7-8aa0-6e0181e9c7d6.png)
![kubuntu-develin](https://user-images.githubusercontent.com/22174615/30583585-fb7d35aa-9d16-11e7-84d6-e6db9f67d5f1.png)

[CC-degree-curve-fit.txt](https://github.com/prasadtalasila/IRCLogParser/files/1313580/CC-degree-curve-fit.txt)
[in-degree-curve-fit.txt](https://github.com/prasadtalasila/IRCLogParser/files/1313581/in-degree-curve-fit.txt)


## Notify reviewers
@rohangoel96

kaushik-rohit commented:
@prasadtalasila I have removed the filter_val from generate_log_plots. WIthout the filter the outputs for CC_degree_plots and k-ubuntu-devel degree plots are attached along with the curve fit parameters. Can you please have a look at this once.

prasadtalasila commented on lib/vis.py:
```diff
@@ -254,8 +254,8 @@ def generate_log_plots(filter_val, plot_data, output_directory, output_file_name
         sum_each_row.append(row)
 
     # print sum_each_row
```
@kaushik-rohit This comment of sum_each_row is misleading. The comment may be intended for line-254, in which case it would either come between lines-253 and 254 or at the end of line-254.

prasadtalasila replied:
@kaushik-rohit This comment of sum_each_row is misleading. The comment may be intended for line-254, in which case it would either come between lines-253 and 254 or at the end of line-254.

prasadtalasila commented on lib/vis.py:
```diff
@@ -232,7 +232,7 @@ def plot_infomap_igraph(nx_graph, membership, output_directory, output_file_name
+def generate_log_plots(plot_data, output_directory, output_file_name):
     """
         Generate log plots for given time frame selecting first filter_val number ofan
         elements and plotting log of value on y axis.
```
please remove the filter_val parameter from sphix documentation.

prasadtalasila reviewed (changes requested):
@kaushik-rohit 
please make the suggested changes.

prasadtalasila commented:
@kaushik-rohit The CC graph is an undirected graph. Performing in and out-degree analysis on an undirected graph does not provide useful information. We should just be doing total degree analysis. 

We have not done a curve fit on the node degree distribution of CC graph. Any new information is welcome though.

kaushik-rohit commented:
@prasadtalasila The in-degree-curve-fit is for kubuntu-devel. The CC degree analysis is for total degree.
The 1st image is CC degree analysis and secondis  exchange degree analysis for kubuntu-devel and like-wise for the curve-fits txt.

kaushik-rohit commented:
@prasadtalasila Also the generate_log_plots has been written in a way that it calls calc_plot_linear_fit for plotting after performing log operation of x_data and y_data. So incase we don't want linear fit for log plots we need to refactor this part.

prasadtalasila commented:
@kaushik-rohit Please launch analysis of Ubuntu channels on the new project server (10.1.9.177) for the month of Jan, 2013. The modified code gets exercised for CC, CU and UU plots. We will be able to make a fair comparison at that point.

prasadtalasila commented:
Also the month long analysis is not likely to give any degree beyond 9 on a log-log plot. After the one month analysis is done, you may also launch a CC analysis on 2013 year on the server.

Better yet, for this kind of scenarios, you will get good quality, fast answers if you put unit tests. Before you launch an analysis, in the sample.py file, put in glue code to pickle the input graphs and the results of functions. If we find the results acceptable, you can then put the picked inputs and outputs into unit tests.

rohangoel96 commented:
@kaushik-rohit  can you please fix the code-climate issues so that we can review and merge?

kaushik-rohit committed:
vis.py: remove upper limit in data plotting

Currently many functions in vis have intial_row_filter
which is being used to determine the data to be used
for plotting. This leaves vis doing more work than just
plotting. The data should be preprocessed outside the vis
and not inside.

Functions with filter ranges:
1) generate_probability_distribution
2) exponential_curve_fit_and_plot
3) exponential_curve_fit_and_plot_x_shifted
4) generate_log_plots
This commit modifies all the above functions and
remove filter parameter.
Rather new functions will be added to determine the correct
percentile cutoff for data selection

Issue Link: #136
Issue Link: #141

kaushik-rohit commented:
@prasadtalasila Sir please merge this after travis build is done.

rohangoel96 commented:
@kaushik-rohit build is failing
